packages/contracts/contracts/vesting/Vesting.sol

```diff
@@ -1,46 +1,51 @@
 import {OwnablePausableUpgradeable} from "../abstracts/OwnablePausableUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/access/AccessControlUpgradeable.sol";
 import {ReentrancyGuardUpgradeable} from "@openzeppelin/contracts-upgradeable/utils/ReentrancyGuardUpgradeable.sol";
 import {TransferHelper} from "@uniswap/lib/contracts/libraries/TransferHelper.sol";
 import {IVesting, IAtomicSwapBase} from "./interfaces/IVesting.sol";
 import {AtomicSwapMsgValidator} from "../abstracts/libs/utils/AtomicSwapMsgValidator.sol";
+import {IERC721} from "@openzeppelin/contracts/token/ERC721/IERC721.sol";
+import {console} from "hardhat/console.sol";
 /// @title Vesting Contract
 /// @notice Implements vesting schedules for token distribution with a cliff period.
 /// @dev Utilizes OpenZeppelin's Ownable and ReentrancyGuard contracts for security.
-contract Vesting is OwnablePausableUpgradeable, ReentrancyGuardUpgradeable, IVesting {
+contract Vesting is OwnablePausableUpgradeable, ReentrancyGuardUpgradeable, IVesting,ERC721Upgradeable {
     using AtomicSwapMsgValidator for *;
     /// @notice Stores vesting schedules for each beneficiary.
-    mapping(address => mapping(bytes32 => VestingSchedule))
+    mapping(uint => VestingSchedule)
         public vestingSchedules;
 
     /// @notice Stores release information for each beneficiary.
     // slither-disable-next-line uninitialized-state
-    mapping(address => mapping(bytes32 => IAtomicSwapBase.Release[]))
+    mapping(uint => IAtomicSwapBase.Release[])
         public releaseInfos;
 
     /// @notice Initializes the vesting contract with necessary parameters.
     /// @param _admin The address of the admin.
     function initialize(address _admin) external initializer {
         __OwnablePausableUpgradeable_init(_admin);
+        __ERC721_init("SideVestingID", "SideVestingID");
     }
     // TODO: Check this function, anyone can call startVesting directly without takeSwap operation
     /// @notice Starts the vesting schedule for a beneficiary.
-    /// @param beneficiary The address of the beneficiary.
+    /// @param buyer The address of the beneficiary.
     /// @param token The token address for vesting.
     /// @param totalAmount The total amount of tokens to be vested.
     /// @param releases Array of release schedules.
     /// @dev Sets up the vesting schedule and release information for the beneficiary.
     function startVesting(
         bytes32 orderId,
-        address beneficiary,
+        address buyer,
         address token,
         uint256 totalAmount,
         IAtomicSwapBase.Release[] memory releases
     ) external payable nonReentrant onlyAdmin {
         // Ensure the uniqueness of 'beneficiary-orderId', to avoid an override call attack.
-        if (vestingSchedules[beneficiary][orderId].from != address(0)) { 
+        uint newVestingId = _issueVestingID(buyer, orderId);
+        if (vestingSchedules[newVestingId].from != address(0)) { 
```

Comment:
what's the value for `vestingSchedules[newVestingId].from`?  token_id of NFT or msg.sender?

Reply:
from is just msg.sender. in this case, it will become atomswap contract address.